src/libcrun/utils.c

```diff
@@ -426,10 +426,10 @@ crun_safe_ensure_at (bool dir, int dirfd, const char *dirpath, size_t dirpath_le
         ret = mkdirat (cwd, cur, mode);
       else
         {
+          cleanup_close int wd_file_cleanup = -1;
```

Comment:
should be moved back as in the original block, otherwise the fd is closed as soon as we leave the else block